client/react/lib/DependenciesProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { FunctionComponent, useContext, useMemo } from 'react';
+import { DynamicoClient } from '@dynamico/core';
+import { DynamicoContext } from './DynamicoProvider';
+import { DependencyOptions, mergeDependencies } from './utils';
+
+export interface DependenciesProviderProps extends Partial<DependencyOptions> {}
+
+export const DependenciesProvider: FunctionComponent<DependenciesProviderProps> = ({ children, ...options }) => {
+  const client = useContext(DynamicoContext);
+  const overrideClient = useMemo(
+    () =>
+      client &&
+      new DynamicoClient({
+        url: client.url,
+        cache: client.cache,
```

Comment:
you might want to add a special prefix here so you won't override global host component versions with ones that match only this context